assume/world.py

```diff
@@ -293,41 +300,43 @@ def setup(
             )
             self.container.register(self.clock_manager)
 
-    def setup_learning(self, episode: int, eval_episode: int) -> None:
+    def setup_learning(
+        self, learning_config: dict, episode: int, eval_episode: int
+    ) -> None:
         """
         Set up the learning process for the simulation, updating bidding parameters with the learning configuration
         and initializing the reinforcement learning (RL) learning role with the specified parameters. It also sets up
         the RL agent and adds the learning role to it for further processing.
         """
 
-        self.bidding_params.update(self.learning_config)
-
-        if self.learning_mode or self.evaluation_mode:
-            # if so, we initiate the rl learning role with parameters
+        # do not create learning role if learning config is missing completely
+        if learning_config:
             from assume.reinforcement_learning.learning_role import Learning
 
-            self.learning_role = Learning(
-                self.learning_config, start=self.start, end=self.end
-            )
-
-            rl_agent = agent_composed_of(
-                self.learning_role,
-                register_in=self.container,
-                suggested_aid="learning_agent",
-            )
-            rl_agent.suspendable_tasks = False
+            # create LearningConfig object
+            learning_config = LearningConfig(**learning_config)
```

Comment:
I would rename the dict to `learning_config_dict` or just `learning_dict` (in world.setup, setup_world, world.setup_learning)- overwriting the same variable with a different type is hard to detect otherwise.

Having the variable named learning_config but it is not of type LearningConfig is confusing